security/hashSaltPassword.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+var crypto = require('crypto');
+
+/* Generates random string of characters i.e salt. */
+var genRandomString = function(length) {
+	return crypto.randomBytes(Math.ceil(length/2))
+		.toString('hex') /** convert to hexadecimal format */
+		.slice(0,length);   /** return required number of characters */
+};
+
+/* Hash password with pbkdf2 and sha512. */
```

Comment:
This is largely the same structure but we'll need usage of 2 functions here: 

For create Acc/Change password -
createNewHash(password) 
=> Will return hash and random gen salt used

For Login Acc-
generateHash(password, salt)
=>will return just hash using salt provided